### ISSUE TITLE
tor-browser-bundle-bin: Unable to connect, does not connect to Tor network

### ISSUE THREAD

**Issue by schmittlauch** [NixOS/nixpkgs#71005]
**Describe the bug**

After upgrading to NixOS release-19.09, I discovered that `tor-browser` from `8hv55zrk16w20pxggx41zq68n05qfari-tor-browser-bundle-bin-8.5.5` is unable to load any websites with the error message "The proxy server is refusing connections".
After further investigations it turns out that **the tor client does not connect to the Tor network**.
When starting the tor-browser build `1fym1h6b3l3319b0nzp93xkc25nkxy71-tor-browser-bundle-bin-8.5.5` from the release-19.03 channel (which was still left in my Nix store), the console output contains info messages of tor as it connects to the tor network. These log messages are entirely missing when using the release-19.09 build.

**To Reproduce**
1. `nix-shell -I nixpkgs=https://github.com/nixos/nixpkgs-channels/archive/nixos-19.09.tar.gz -p tor-browser-bundle-bin --run 'tor-browser'`
1. check the console output for any tor-related messages
1. close tor-browser
1. try to visit a website in Tor browser e.g. https://nixos.org/
1. `nix-shell -I nixpkgs=https://github.com/nixos/nixpkgs-channels/archive/nixos-19.03.tar.gz -p tor-browser-bundle-bin --run 'tor-browser'`
1. check the console output for any tor-related messages
1. try to visit a website in Tor browser e.g. https://nixos.org/

**Expected behavior**

At the start of tor-browser, the console logging messages are expected to contain tor connection messages like 

```
Oct 11 23:18:03.636 [notice] Tor 0.4.1.5 (git-439ca48989ece545) running on Linux with Libevent 2.1.8-stable, OpenSSL 1.0.2s, Zlib 1.2.11, Liblzma N/A, and Libzstd N/A.
Oct 11 23:18:03.636 [notice] Tor can't help you if you use it wrong! Learn how to be safe at https://www.torproject.org/download/download#warning
Oct 11 23:18:03.636 [notice] Read configuration file "/nix/store/1fym1h6b3l3319b0nzp93xkc25nkxy71-tor-browser-bundle-bin-8.5.5/share/tor-browser/TorBrowser/Data/Tor/torrc-defaults".
```

The tor-browser Firefox component shall be able to retrieve websites.

With the release-19.09 build, tor connection log messages are entirely missing and wbesites do not load with "The proxy server is refusing connections"

**Additional context**

channel state: `19.09.716.88bbb3c8096`

`tor-browser-bundle` (without `-bin`) is affected as well on release-19.09

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.5, NixOS, 19.09.716.88bbb3c8096 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"home-manager-19.09, nixos-19.09.716.88bbb3c8096, nixos-unstable-20.03pre196201.07d4df59626"`
 - channels(user): `"home-manager-19.03"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: [ tor-browser-bundle-bin, tor-browser-bundle ]
# a list of nixos modules affected by the problem
module:
```
pinging maintainers: @offline @matejc @doublec @thoughtpolice @joachifm

**Comment by joachifm** [comment]
I'm unable to reproduce this. Can you try running `TBB_HOME=$(mktemp -d) tor-browser` to rule out the possibility of corrupt/stale profile (this should not happen but still ...).

**Comment by schmittlauch** [comment]
This indeed seems to be caused by corrupt state: Deleting `~/.local/share/tor-browser/TorBrowser/` resolved this issue.
Until someone else complains about a similar issue, I assume I'm the only one affected.

**Comment by schmittlauch** [comment]
I have to re-open this issue. I am reliably able to reproduce this issue by

1. start with a fresh state directory: `rm ~/.local/share/tor-browser/TorBrowser -rf`
1. start `tor-browser`, connect to tor, load a website, close the window
1. start `tor-browser` again, load a website, close the window
1. start `tor-browser` again, load a website. Now the issue described above appears.

So for me tor isn't started anymore the third time I launch tor-browser.

**Comment by Ma27** [comment]
I can reproduce the issue as well. Will try to look into this :)

**Comment by schmittlauch** [comment]
This seems to be fixed with tor-browser-bundle-bin-9.0.1 in release-19.09.

/cc @l0b0
